xdist/dsession.py

```diff
@@ -120,6 +120,10 @@ def pytest_runtestloop(self):
     def loop_once(self):
         """Process one callback from one of the slaves."""
         while 1:
+            if not self._active_nodes:
+                # If everything has died stop looping
+                self.triggershutdown()
+                raise RuntimeError("No active nodes")
```

Comment:
i propose something more along the lines of ``unexpectedly no active workers available`` so people can understand the error more easyly

Reply:
Fixed.